tests/backends/test_native_backend_against_coincurve.py

```diff
@@ -84,33 +84,53 @@ def test_signing_is_equal(private_key_bytes,
-        st.just('coincurve-to-native'),
-        st.just('native-to-coincurve'),
-    ),
 )
 @settings(max_examples=MAX_EXAMPLES)
-def test_native_to_coincurve_recover(private_key_bytes,
+def test_coincurve_to_native_recover(private_key_bytes,
                                      message_hash,
-                                     direction,
                                      native_key_api,
                                      coincurve_key_api):
-    if direction == 'coincurve-to-native':
-        backend_a = coincurve_key_api
-        backend_b = native_key_api
-    elif direction == 'native-to-coincurve':
-        backend_b = coincurve_key_api
-        backend_a = native_key_api
+
+    coincurve_private_key = coincurve_key_api.PrivateKey(private_key_bytes)
+    coincurve_public_key = coincurve_private_key.public_key
+    coincurve_signature = coincurve_key_api.ecdsa_sign(message_hash, coincurve_private_key)
+
+    # The hypothesis message_hash_st shrinks the message hash,
+    # so it tests the zero byte message hash every time.
+    # If message_hash == b'\x00' * 32, the corresponding Jacobian point
+    # on the curve is the point at infinity.
+    # The native backend raises a BadSignature error on ecdsa_recover
+    # for the point at infinity,
+    # whereas the coincurve library returns an address.
+    # This test does introduce some flakiness on the small chance
+    # that hypothesis generates another message hash that corresponds to the
+    # Jacobian point at infinity
+    if message_hash == b'\x00' * 32:
+        with pytest.raises(BadSignature):
+            native_key_api.ecdsa_recover(message_hash, coincurve_signature)
     else:
-        assert False, "invariant"
+        native_public_key = native_key_api.ecdsa_recover(message_hash, coincurve_signature)
+        assert native_public_key == coincurve_public_key
+        assert native_key_api.ecdsa_verify(message_hash, coincurve_signature, native_public_key)
 
-    private_key_a = backend_a.PrivateKey(private_key_bytes)
-    public_key_a = private_key_a.public_key
-    signature_a = backend_a.ecdsa_sign(message_hash, private_key_a)
 
-    public_key_b = backend_b.ecdsa_recover(message_hash, signature_a)
-    assert public_key_b == public_key_a
-    assert backend_b.ecdsa_verify(message_hash, signature_a, public_key_b)
+@given(
+    private_key_bytes=private_key_st,
+    message_hash=message_hash_st,
+)
+@settings(max_examples=MAX_EXAMPLES)
+def test_native_to_coincurve_recover(private_key_bytes,
+                                     message_hash,
+                                     native_key_api,
+                                     coincurve_key_api):
+
+    native_private_key = native_key_api.PrivateKey(private_key_bytes)
+    native_public_key = native_private_key.public_key
+    native_signature = native_key_api.ecdsa_sign(message_hash, native_private_key)
+
+    coincurve_public_key = coincurve_key_api.ecdsa_recover(message_hash, native_signature)
```

Comment:
Hm, shouldn't the coincurve backend also complain about the point at infinity? This makes it look like we only fixed the bug when using the native back end.